File: languageBenchmark/Collatz.go
package main

import (
	"fmt"
	"strconv"
	"time"
	"os"
)

func main() {
	startTime := time.Now()
	count := 1
	answer := 1
	broken := false
	file, _ := os.Create("go.txt")

	for broken != true {
		for answer != 1 {
			if answer % 2 == 1 {
				answer = (answer * 3) + 1
			} else if answer <= 0 {
				fmt.Println(strconv.Itoa(count) + " solved the Collatz conjecture!")
				os.Exit(1)
			} else {
				answer = answer/2
			}
		}
		count = count + 1
		answer = count
		if count % 1000000 == 0 { // record every 1 million
			fmt.Println("Reached " + strconv.Itoa(count) + " in " + time.Now().Sub(startTime).String())
			file.WriteString(strconv.Itoa(count) + "," + time.Now().Sub(startTime).String() + "\n")
		}
	}

}

File: languageBenchmark/Collatz.py
import sys
import time

def main():
    startTime = time.time()
    count = 1
    answer = 1
    broken = False
    file = open("python.txt", "w+")

    while not broken:
        while answer != 1:
            if answer % 2 == 1:
                answer = (answer * 3) + 1
            elif answer <= 0:
                print(str(count) + " solved the Collatz conjecture!")
                sys.exit()

            else:
                answer = answer/2

        count = count + 1
        answer = count
        if count % 100000 == 0:
            print("Reached " + str(count) + " in "+ str(time.time() - startTime) + " seconds.") # print updates
            file.write(str(count) + "," + str(time.time() - startTime) + "\n") # write to a file


if __name__ == '__main__':
    main()